src/main/java/io/toast/tk/adapter/web/component/WebTableElement.java

```diff
@@ -0,0 +1,99 @@
+package io.toast.tk.adapter.web.component;
+
+import java.util.List;
+
+import org.apache.commons.lang.NullArgumentException;
+import org.apache.commons.lang3.StringUtils;
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebElement;
+
+import io.toast.tk.automation.driver.web.SeleniumSynchronizedDriver;
+import io.toast.tk.core.runtime.IWebElementDescriptor;
+
+public class WebTableElement extends WebAutoElement {
+
+	private static final String ROW_VALUE_SEPARATOR = "|";
+
+	public WebTableElement(
+			IWebElementDescriptor element,
+			SeleniumSynchronizedDriver driver) {
+		super(element, driver);
+	}
+
+	public WebTableElement(
+			IWebElementDescriptor element) {
+		super(element);
+	}
+
+	public int countLines() throws IllegalAccessException {
+		WebElement table = getTableWebElement();
+		List<WebElement> rows = table.findElements(By.tagName("tr"));
+		return rows.size();
+	}
+
+	private WebElement getTableWebElement() throws IllegalAccessException {
+		WebElement table = frontEndDriver.find(descriptor);
+
+		if (table == null) {
+			throw new NullArgumentException("WebElement table is null");
+		} else if (table.getTagName().toLowerCase().equals("table")) {

```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Move the "table" string literal on the left side of this string comparison. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.talanlabs.com/coding_rules#rule_key=squid%3AS1132)
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Replace these toUpperCase()/toLowerCase() and equals() calls with a single equalsIgnoreCase() call. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.talanlabs.com/coding_rules#rule_key=squid%3AS1157)